src/function/cast/blob_cast.cpp

```diff
@@ -11,6 +11,8 @@ BoundCastInfo DefaultCasts::BlobCastSwitch(BindCastInput &input, const LogicalTy
 		return BoundCastInfo(&VectorCastHelpers::StringCast<string_t, duckdb::CastFromBlob>);
 	case LogicalTypeId::AGGREGATE_STATE:
 		return DefaultCasts::ReinterpretCast;
+	case LogicalTypeId::UNION:
+		return DefaultCasts::ImplicitToUnionCast(input, source, target);
```

Comment:
Can we add this to the `default_casts.cpp` once instead of to all of the separate casts?

Reply:
Ok, there are some special cases though, UNION -> UNION and NULL -> UNION does not use the same ImplicitToUnionCast, so we have to guard against them as well and can't just do a `if(target.id() == LogicalTypeId::UNION) { ImplicitToUnionCast(...) }`